api/types/secretargs.go

```diff
@@ -16,4 +16,10 @@ type SecretArgs struct {
 	// If type is "kubernetes.io/tls", then "literals" or "files" must have exactly two
 	// keys: "tls.key" and "tls.crt"
 	Type string `json:"type,omitempty" yaml:"type,omitempty"`
+
+	// EmitStringData if true generates a v1/Secret with plain-text stringData fields
+	// instead of base64-encoded data fields. If a generating field does not have a
+	// UTF-8 value, it falls back to being stored as a base64-encoded data field. This
+	// is similar to the default binaryData fallback of a ConfigMapGenerator.
```

Comment:
```suggestion
	// EmitStringData requests generating a v1/Secret with plain-text "stringData" fields
	// instead of base64-encoded "data" fields. If a would-be "stringData" field's value
	// is not valid UTF-8, it falls back to emitting a base64-encoded "data" field instead,
	// analogously to the "binaryData" fallback of a ConfigMapGenerator.
```